internal/bgp/messages.go

```diff
@@ -301,12 +301,22 @@ func sendUpdate(w io.Writer, asn uint32, ibgp, fbasn bool, defaultNextHop net.IP
 		return err
 	}
 	l := b.Len()
-	if err := encodePathAttrs(&b, asn, ibgp, fbasn, defaultNextHop, adv); err != nil {
-		return err
+	if adv.Prefix.IP.To4() != nil {
+        if err := encodePathAttrs(&b, asn, ibgp, fbasn, defaultNextHop, adv); err != nil {
+            return err
+        }
+        binary.BigEndian.PutUint16(b.Bytes()[21:23], uint16(b.Len()-l))
+        encodePrefixes(&b, []*net.IPNet{adv.Prefix})
+        binary.BigEndian.PutUint16(b.Bytes()[16:18], uint16(b.Len()))
+	} else if adv.Prefix.IP.To16() != nil {
+        if err := encodePathAttrsIPv6(&b, asn, ibgp, fbasn, defaultNextHop, adv); err != nil {
+            return err
+        }
+        binary.BigEndian.PutUint16(b.Bytes()[21:23], uint16(b.Len()-l))
+        binary.BigEndian.PutUint16(b.Bytes()[16:18], uint16(b.Len()))
+	} else {
+	    return nil
```

Comment:
Why nil here? If `adv.Prefix.IP.To4()` and `adv.Prefix.IP.To16()` fail, don't we want to throw an error?

Reply:
after removing the explicit check for ipv4 some announcements where getting through with an IP of nil. It seemed harmless to discard them.